src/main/java/tatracker/ui/MainWindow.java

```diff
@@ -229,16 +282,10 @@ private CommandResult executeCommand(String commandText) throws CommandException
             CommandResult commandResult = logic.execute(commandText);
             logger.info("Result: " + commandResult.getFeedbackToUser());
             resultDisplay.setFeedbackToUser(commandResult.getFeedbackToUser());
-
-            if (commandResult.isShowHelp()) {
-                handleHelp();
-            }
-
-            if (commandResult.isExit()) {
-                handleExit();
-            }
+            handleCommand(commandText);
```

Comment:
Err as in not handle the command text. You should change command result to have a hasSwitchedTab and a getSwitchedTab. The idea is not to handle the word "goto" in this way...

Reply:
Rather than the Main Window I mean.